### ISSUE TITLE
Error when using RETURN with UPDATE REMOVE

### ISSUE THREAD

**Issue by Ziink** [orientechnologies/orientdb#4251]
``` SQL
update #17:16 remove note
```

works as expected. But

``` SQL
update #17:16 remove note return after
```

throws

com.orientechnologies.orient.core.sql.OCommandSQLParsingException: Error on parsing command at position #25: Found unexpected keyword 'RETURN' while it was expected '[=, WHERE]'. Use UPDATE <class>|cluster:<cluster>> [SET|ADD|PUT|REMOVE|INCREMENT|CONTENT {<JSON>}|MERGE {<JSON>}] [[,] <field-name> = <expression>|<sub-command>]\* [LOCK <NONE|RECORD>] [UPSERT] [RETURN <COUNT|BEFORE|AFTER>] [WHERE <conditions>] Command: UPDATE #17:16 REMOVE note RETURN AFTER ---------------------------------^

orientdb-community-2.1-rc3

**Comment by Nirlah** [comment]
Yes, happens to me to [2.1.0].

I think the parser is strictly expecting `= someValue` before the `return` although it is not required by the syntax definition.

@lvca please add to a hotfix milestone so this issue won't be lost.

**Comment by Eric24** [comment]
I wonder is this is related? The following UPDATE with a PUT also throws a similar exception, but works if I replace the PUT with a SET on a non-map property:

`UPDATE #14:1 PUT Options = '22', {"Action":"Test22a","Enabled":false, "@type":"d", "@class":"VOptions"} WHERE NOT (Options CONTAINSKEY '22') RETURN AFTER @this.Options['22']
`

**Comment by luigidellaquila** [comment]
This is fixed in v 3.0 with the new SQL executor

Closing, thanks

Luigi
